src/helpers/makeLinkedHeader.js

```diff
@@ -41,8 +42,7 @@ export const makeLinkedHeader = (
 
   const WrappedComponent = (props) => {
     const { children } = props;
-    const id =
-      props.id || slugify(Array.isArray(children) ? children.join() : children);
+    const id = props.id || slugify(loopAndExtractString(children));
```

Comment:
We just hadn't hit this bug, but it was waiting to happen. Use the recursive string extraction helper to build this `id`